<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>901</width>
    <height>807</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="4">
    <widget class="QListView" name="listView_props"/>
   </item>
   <item row="7" column="0" colspan="5">
    <widget class="QListView" name="listView_results"/>
   </item>
   <item row="0" column="3">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Motors considered</string>
     </property>
    </widget>
   </item>
   <item row="0" column="4">
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>Propellers considered</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="plane_label">
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt; font-weight:600;&quot;&gt;TextLabel&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
   </item>
   <item row="6" column="4">
    <widget class="QPushButton" name="pushButton_plot_performance">
     <property name="toolTip">
      <string>Show performance map of combination</string>
     </property>
     <property name="text">
      <string>Plot combination performance</string>
     </property>
    </widget>
   </item>
   <item row="1" column="3">
    <widget class="QListView" name="listView_motors"/>
   </item>
   <item row="2" column="3" rowspan="4" colspan="2">
    <widget class="QWidget" name="widget_plot" native="true">
     <property name="minimumSize">
      <size>
       <width>500</width>
       <height>500</height>
      </size>
     </property>
     <layout class="QVBoxLayout" name="layout_plot"/>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="label_4">
     <property name="text">
      <string>Optimising plane:</string>
     </property>
    </widget>
   </item>
   <item row="6" column="3">
    <widget class="QPushButton" name="pushButton_plot_efficiency">
     <property name="toolTip">
      <string>Show efficiency map of combination</string>
     </property>
     <property name="text">
      <string>Plot combination efficiency</string>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QPushButton" name="pushButton_get_optimal">
     <property name="toolTip">
      <string>Calculate the best combinations from the selection</string>
     </property>
     <property name="text">
      <string>Find optimal combinations</string>
     </property>
    </widget>
   </item>
   <item row="5" column="1">
    <widget class="QLabel" name="label_3">
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;right&quot;&gt;Top&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
   </item>
   <item row="8" column="3">
    <widget class="QPushButton" name="pushButton_return">
     <property name="toolTip">
      <string>View selected results in other window</string>
     </property>
     <property name="text">
      <string>Return to analyser and view selected results</string>
     </property>
    </widget>
   </item>
   <item row="8" column="4">
    <widget class="QPushButton" name="pushButton_cancel">
     <property name="text">
      <string>Cancel</string>
     </property>
    </widget>
   </item>
   <item row="4" column="0" colspan="3">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Constraints</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="1" column="1">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_max_current">
        <property name="maximum">
         <double>500.000000000000000</double>
        </property>
        <property name="value">
         <double>110.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QCheckBox" name="checkBox_static_Th_const">
        <property name="toolTip">
         <string>Ignore results which produce less than a required thrust at take off (for hand launching)</string>
        </property>
        <property name="text">
         <string>Take off thrust</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QCheckBox" name="checkBox_motor_max_mass_const">
        <property name="toolTip">
         <string>Ignore results with a motor heavier than a given mass</string>
        </property>
        <property name="text">
         <string>Motor weight</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_top_speed">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_motor_max_mass">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="maximum">
         <double>1000.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="checkBox_top_speed_const">
        <property name="toolTip">
         <string>Ignore results that fail to reach a top speed</string>
        </property>
        <property name="text">
         <string>Top speed</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="checkBox_max_current_const">
        <property name="toolTip">
         <string>Ignore results that result in currents higher than a given percentage of the motors rated current</string>
        </property>
        <property name="text">
         <string>Max. allowable current</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_static_Th">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>N</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>%</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>m/s</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>g</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0" colspan="3">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Optimise at speed</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="1" column="2">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>m/s</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QRadioButton" name="radioButton_fixed_speed">
        <property name="toolTip">
         <string>Results are calculated at a given fixed flight speed</string>
        </property>
        <property name="text">
         <string>Fixed flight speed</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="3">
       <widget class="QRadioButton" name="radioButton_optimal_speed">
        <property name="toolTip">
         <string>Results are calculated using the optimal speed for each combination</string>
        </property>
        <property name="text">
         <string>Optimal flight speed</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_flight_speed">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="value">
         <double>10.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0" colspan="3">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Optimisation type</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QRadioButton" name="radioButton_maximise_time">
        <property name="toolTip">
         <string>Rank results by total flight time</string>
        </property>
        <property name="text">
         <string>Flight duration</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="radioButton_maximise_distance">
        <property name="toolTip">
         <string>Rank results by distance flown</string>
        </property>
        <property name="text">
         <string>Distance flown</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="5" column="2">
    <widget class="QSpinBox" name="spinBox_no_best">
     <property name="toolTip">
      <string>Number of results to return</string>
     </property>
     <property name="maximum">
      <number>20</number>
     </property>
     <property name="value">
      <number>5</number>
     </property>
    </widget>
   </item>
   <item row="6" column="0">
    <widget class="QLabel" name="label_10">
     <property name="text">
      <string>Number of motors</string>
     </property>
    </widget>
   </item>
   <item row="6" column="1" colspan="2">
    <widget class="QSpinBox" name="spinBox_no_motors">
     <property name="value">
      <number>1</number>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
